app/controllers/application_controller.rb

```diff
@@ -9,6 +9,7 @@ class ApplicationController < ActionController::Base
   before_action :set_current_user
   before_action :redirect_unauthenticated
   before_action :access_control
+  before_action :primary_account
```

Comment:
To be fair, `:access_control` just above wasn't a great example, but it's nice to phrase methods like these as an imperative, e.g. `check_primary_account`, `handle_subsidiary_accounts`, etc.